Quick_Deploy/vLLM/model_repository/vllm/1/model.py

```diff
@@ -0,0 +1,330 @@
+from vllm.outputs import RequestOutput as VLLMOutput
+
+_VLLM_ENGINE_ARGS_FILENAME = "vllm_engine_args.json"
+
+
+class VLLMAsyncEngineConfig(BaseModel):
+    # required model
+    model: str
+
+    # arguments from vLLM engine
+    max_num_batched_tokens: int = 2560
+    max_num_seqs: int = 256
+    disable_log_requests: bool = True
+
+
+class CompletionRequest(BaseModel):
+    # required prompt
+    prompt: str
+
+    # generation parameters
+    max_tokens: Optional[int] = 16
+    stop: Optional[Union[str, List[str]]] = None
+
+    # sampling parameters
+    temperature: Optional[float] = 1.0
+    top_k: Optional[int] = -1
+    top_p: Optional[float] = 1.0
+    ignore_eos: Optional[bool] = False
+
+    # output options
+    echo: Optional[bool] = False
+    stream: Optional[bool] = False
+
+
+class Completion(BaseModel):
+    index: int
+    text: str
+    gen_token_count: Optional[int] = None
+    cumulative_logprob: Optional[float] = None
+    finish_reason: Optional[Literal["stop", "length"]] = None
+
+
+class CompletionResponse(BaseModel):
+    request_id: Optional[str] = None
+    finished: bool = False
+
+    prompt: Optional[str] = None
+    prompt_token_count: Optional[int] = None
+
+    completions: List[Completion] = []
+
+    # inflight request stats
+    current_inflight_count: Optional[int] = None
+    average_inflight_count: Optional[float] = None
+
+
+class OutputOptions:
+    def __init__(self, *, echo: bool = False, stream: bool = False):
+        self._echo = echo
+        self._stream = stream
+
+    @property
+    def echo(self) -> bool:
+        """
+        If true, the prompt and prompt token ids will be echoed back to the client in the response.
+        """
+        return self._echo
+
+    @property
+    def stream(self) -> bool:
+        """
+        If true, the response will be streamed back to the client as it is generated.
+        """
+        return self._stream
+
+
+class TritonPythonModel:
+    def initialize(self, args):
+        self.logger = pb_utils.Logger
+
+        # load triton model config from args
+        self.model_config = json.loads(args["model_config"])
+
+        # assert are in decoupled mode
+        self.using_decoupled = pb_utils.using_decoupled_model_transaction_policy(
+            self.model_config
+        )
+        assert (
+            self.using_decoupled
+        ), "vLLM Triton backend must be configured to use decoupled model transaction policy"
+
+        engine_args_filepath = os.path.join(
+            args["model_repository"], _VLLM_ENGINE_ARGS_FILENAME
+        )
+        assert os.path.isfile(
+            engine_args_filepath
+        ), f"'{_VLLM_ENGINE_ARGS_FILENAME}' containing vllm engine args must be provided in '{args['model_repository']}'"
+        with open(engine_args_filepath) as file:
+            vllm_engine_config = VLLMAsyncEngineConfig(**json.load(file))
+
+        # create the vllm engine
+        engine_args = AsyncEngineArgs(
```

Comment:
Instead of creating our own class for configs, can we use the json dictionary directly?

That way we don't have to change to add aditional config options - 
```
engine_args = AsyncEngineArgs(**json.load(file))
```